packages/cardano-graphql-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/mappers.ts

```diff
@@ -1,16 +1,150 @@
 import { Cardano } from '@cardano-sdk/core';
-import { PoolHash, PoolHashModel } from './types';
+import {
+  EpochRewardModel,
+  OwnerAddressModel,
+  PoolData,
+  PoolDataModel,
+  PoolOwner,
+  PoolRegistration,
+  PoolRegistrationModel,
+  PoolRelay,
+  PoolRetirement,
+  PoolRetirementModel,
+  PoolUpdate,
+  PoolUpdateModel,
+  RelayModel
+} from './types';
 
 const toHexString = (bytes: Buffer) => bytes.toString('hex');
 
-// TODO: create Cardano.StakePool object by grouping fields using hashes and update ids
-// TODO: create interface for StakePoolFields
-export const toCoreStakePool = (stakePoolFields: any): Cardano.StakePool[] => [];
+const getPoolStatus = (
+  lastPoolRegistration: PoolRegistration,
+  lastEpoch: number,
+  lastPoolRetirement?: PoolRetirement
+) => {
+  if (lastPoolRetirement === undefined || lastPoolRetirement.retiringEpoch <= lastPoolRegistration.activeEpochNo) {
+    if (lastPoolRegistration.activeEpochNo > lastEpoch) return Cardano.StakePoolStatus.Activating;
+    return Cardano.StakePoolStatus.Active;
+  }
+  if (lastPoolRetirement.retiringEpoch > lastEpoch) return Cardano.StakePoolStatus.Retiring;
+  return Cardano.StakePoolStatus.Retired;
+};
 
-export const mapPoolHash = (poolHashModel: PoolHashModel): PoolHash => ({
-  hash: toHexString(poolHashModel.hash_raw),
-  id: poolHashModel.id,
-  view: poolHashModel.view
+interface ToCoreStakePoolInput {
+  poolOwners: PoolOwner[];
+  poolDatas: PoolData[];
+  poolRegistrations: PoolRegistration[];
+  poolRelays: PoolRelay[];
+  poolRetirements: PoolRetirement[];
+  lastEpoch: number;
+}
+
+export const toCoreStakePool = ({
+  poolOwners,
+  poolDatas,
+  poolRegistrations,
+  poolRelays,
+  poolRetirements,
+  lastEpoch
+}: ToCoreStakePoolInput): Cardano.StakePool[] =>
+  poolDatas.map((poolData) => {
+    const registrations = poolRegistrations.filter((r) => r.hashId === poolData.hashId);
+    const retirements = poolRetirements.filter((r) => r.hashId === poolData.hashId);
+    return {
+      cost: poolData.cost,
+      epochRewards: [],
+      hexId: poolData.hexId,
+      id: poolData.id,
+      // FIXME: this values cannot be taken by db sync
```

Comment:
To convert floating point into fraction you could use [fraction.js](https://www.npmjs.com/package/fraction.js)